lib/class-wp-rest-annotations-controller.php

```diff
@@ -0,0 +1,1097 @@
+	 *
+	 * @see WP_REST_Comments_Controller::get_items_permissions_check()
+	 */
+	public function get_items_permissions_check( $request ) {
+		$post_ids       = (array) $request['post'];
+		$types          = (array) $request['type'];
+		$statuses       = (array) $request['status'];
+		$default_params = $request->get_default_params();
+
+		if ( ! $post_ids ) {
+			return new WP_Error( 'rest_missing_annotation_posts', __( 'Missing post(s).', 'gutenberg' ), array(
+				'status' => 400,
+			) );
+		}
+
+		if ( ! $types ) {
+			return new WP_Error( 'rest_missing_annotation_types', __( 'Missing type(s).', 'gutenberg' ), array(
+				'status' => 400,
+			) );
+		}
+
+		if ( ! $statuses ) {
+			return new WP_Error( 'rest_missing_annotation_statuses', __( 'Missing status(es).', 'gutenberg' ), array(
+				'status' => 400,
+			) );
+		}
+
+		foreach ( $post_ids as $post_id ) {
+			if ( ! $post_id ) {
+				return new WP_Error( 'rest_missing_annotations_post', __( 'Missing post.', 'gutenberg' ), array(
+					'status' => 400,
+				) );
+			}
+
+			if ( ! $this->check_read_post_permission( $post_id, $request ) ) {
+				return new WP_Error( 'rest_cannot_read_annotations_post', __( 'Sorry, you are not allowed as this user.', 'gutenberg' ), array(
+					'status' => rest_authorization_required_code(),
+				) );
+			}
+
+			foreach ( $types as $type ) {
+				if ( ! current_user_can( 'read_annotations', $post_id, $type ) ) {
+					return new WP_Error( 'rest_cannot_read_annotations', __( 'Sorry, you are not allowed as this user.', 'gutenberg' ), array(
+						'status' => rest_authorization_required_code(),
+					) );
+				}
+			}
+		}
+
+		foreach ( $types as $type ) {
+			// @codeCoverageIgnoreStart
+			// Not possible to test at this time.
+			// Strengthens security in custom post types and on sites with permissions mods.
+			if ( ! current_user_can( 'edit_annotations', $type ) ) {
+				if ( 'edit' === $request['context'] ) {
+					return new WP_Error( 'rest_forbidden_annotations_context', __( 'Sorry, you are not allowed as this user.', 'gutenberg' ), array(
+						'status' => rest_authorization_required_code(),
+					) );
+				}
+
+				$allow_statuses = array( 'approved', 'approve', '1' );
+				$allow_statuses = array_merge( $allow_statuses, WP_Annotation_Utils::$custom_statuses );
+
+				if ( array_diff( $statuses, $allow_statuses ) ) {
+					return new WP_Error( 'rest_forbidden_annotation_param_status', __( 'Sorry, you are not allowed as this user.', 'gutenberg' ), array(
+						'status' => rest_authorization_required_code(),
+					) );
+				}
+			} // @codeCoverageIgnoreEnd
+
+			if ( ! current_user_can( 'edit_others_annotations', $type ) ) {
+				$current_user_id       = get_current_user_id();
+				$current_user_can_edit = current_user_can( 'edit_annotations', $type );
+				$is_current_user_query = $current_user_id && array( $current_user_id ) === $request['author'];
+
+				foreach ( array(
+					'author',
```

Comment:
The intention here is to prevent non-administrative users (e.g., authors/contributors) from looking at another user's history. The `author`, `author_exclude`, and `author_email` parameters are also guarded in core, [in a similar way, for comments](https://github.com/WordPress/WordPress/blob/b9944def27bc8128d7d90f2d22ba915a3d959677/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php#L138). A useful exception to this rule is that any user who can edit, can list their own annotations; i.e., note the `$is_current_user_query` variable a few lines below this.

Reply:
That makes sense. I think it warrants an inline comment.